src/components/SearchBar/index.tsx

```diff
@@ -15,42 +17,68 @@ interface SearchBarProps {
   validatorMessage?: string;
   className?: string;
   onSubmit?: (selected: string, value: string) => void;
+  onChange?: (selected: string, value: string) => void;
+  shouldClearOnSubmit?: (selected: string) => boolean;
 }
 
 const SearchBar: React.FC<SearchBarProps> = ({
+  className,
   disabled = false,
+  onChange,
+  onSubmit,
+  shouldClearOnSubmit = () => true,
   validator = () => true,
   validatorMessage = "Invalid Input",
-  className,
-  onSubmit,
 }) => {
   const [input, setInput] = useState("");
```

Comment:
I wonder if we should maintain two different inputs for Filter and Search... 🤔

for example, if you do the following:
1. apply a search
2. switch to filter and apply a filter
3. go back to search

the search term is gone from the input but the search is still being applied

Reply:
I replicated legacy behavior we now clear the search after you apply a filter.